### PR TITLE
Define keymappings using a mapping table

### PR DESCRIPTION
**Pull request by Tastyep**
Define a table of mappings like whichkey does, and implement a function for registering them.
This is just a readability improvement, nothing changes apart from the style.

### CODE REVIEW

**Review by rebuilt — APPROVED** [review]
This looks great.  Nice cleanup. 

I confirmed all mapping are present by comparing diffs of the :map command(sorted).  
 :enew|pu=execute('map') 

I also tested most of the keybind to make sure they still work and they do.

**Comment by rebuilt** [comment]
The quickfix toggle doesn't work but there's a pull request for that open:  https://github.com/ChristianChiarulli/LunarVim/pull/752

**Comment by abzcoding** [comment]
@Tastyep thank you for this lovely PR, can you add the #757 settings to this table style as well

**Comment by abzcoding** [comment]
@Tastyep thank you for this PR, can you add the terminal window navigation keybindings to this table as well?
something like
```lua
-- Terminal window navigation
t = {
  { "<C-h>", "<C-\\><C-N><C-w>h"},
  { "<C-j>", "<C-\\><C-N><C-w>j"},
  { "<C-k>", "<C-\\><C-N><C-w>k"},
  { "<C-l>", "<C-\\><C-N><C-w>l"},
  { "<Esc>", "<C-\\><C-n>"},

},
i = {
  { "<C-h>", "<C-\\><C-N><C-w>h"},
  { "<C-j>", "<C-\\><C-N><C-w>j"},
  { "<C-k>", "<C-\\><C-N><C-w>k"},
  { "<C-l>", "<C-\\><C-N><C-w>l"},
}
```

File: lua/keymappings.lua [hunk 1]
```diff
@@ -1,86 +1,108 @@
...
+  n = {
+    -- better window movement
+    { "<C-h>", "<C-w>h", { silent = true } },
+    { "<C-j>", "<C-w>j", { silent = true } },
+    { "<C-k>", "<C-w>k", { silent = true } },
+    { "<C-l>", "<C-w>l", { silent = true } },
 
--- TODO fix this
--- resize with arrows
-if vim.fn.has("mac") == 1 then
```
Comment (colutti) [inline lua/keymappings.lua:23]: You removed the if that control the mappings for macs.

File: lua/keymappings.lua [hunk 2]
```diff
@@ -1,86 +1,108 @@
...
+    -- QuickFix
+    { "]q", ":cnext<CR>" },
+    { "[q", ":cprev<CR>" },
 
--- Move selected line / block of text in visual mode
-vim.api.nvim_set_keymap("x", "K", ":move '<-2<CR>gv-gv", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("x", "J", ":move '>+1<CR>gv-gv", { noremap = true, silent = true })
+    -- {'<C-TAB>', 'compe#complete()', {noremap = true, silent = true, expr = true}},
+  },
+  v = {
```
Comment (colutti) [inline lua/keymappings.lua:39]: Those variables should be named better than this.

**Review by colutti — CHANGES_REQUESTED** [review]
There are some changes I would do, specially the IF that controls keybinds for MACs

File: lua/keymappings.lua (hunk 2 above)
Comment (colutti) [reply lua/keymappings.lua:39]: Like this v =

File: lua/keymappings.lua (hunk 1 above)
Comment (Tastyep) [reply lua/keymappings.lua:23]: It is line 87

File: lua/keymappings.lua (hunk 2 above)
Comment (Tastyep) [reply lua/keymappings.lua:39]: I don't understand what you mean, what is wrong here ?

Comment (colutti) [reply lua/keymappings.lua:39]: the name you gave to the variables, like "v" and others. It could be a little more descriptive like "visual" for example.

Comment (Tastyep) [reply lua/keymappings.lua:39]: The goal is to follow the same pattern as which-key, also understand that these keys are used as parameters for nvim_set_keymap

File: lua/keymappings.lua (hunk 1 above)
Comment (colutti) [reply lua/keymappings.lua:23]: sorry, i missed that.

File: lua/keymappings.lua (hunk 2 above)
Comment (Tastyep) [reply lua/keymappings.lua:39]: As a compromise, I'll add a small comment nest to the key

Comment (colutti) [reply lua/keymappings.lua:39]: ah ok, sounds good

**Review by colutti — APPROVED** [review]

**Review by jameswalmsley — APPROVED** [review]

**Review by abzcoding — APPROVED** [review]

File: lua/keymappings.lua [hunk 3]
```diff
@@ -1,83 +1,106 @@
...
+  t = { -- Terminal mode
+    -- Terminal window navigation
+    { "<C-h>", "<C-\\><C-N><C-w>h" },
+    { "<C-j>", "<C-\\><C-N><C-w>j" },
+    { "<C-k>", "<C-\\><C-N><C-w>k" },
+    { "<C-l>", "<C-\\><C-N><C-w>l" },
+  },
+  v = { -- Visual/Select mode
+    -- Better indenting
+    { "<", "<gv" },
```
Comment (abzcoding) [inline lua/keymappings.lua:64]: I think this is needed as well
`{ noremap = true, silent = true }`

Comment (Tastyep) [reply lua/keymappings.lua:64]: These are the default options passed to register_mappings

Comment (abzcoding) [reply lua/keymappings.lua:64]: :facepalm:  my bad

**Pull request merged**